### PR TITLE
Eager models

### PR DESCRIPTION
**Pull request by dirkgr**

### CODE REVIEW

File: tango/integrations/torch/train.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ class TorchTrainStep(Step):
 
     def run(  # type: ignore[override]
         self,
-        model: Lazy[Model],
+        model: Union[Lazy[Model], Model],  # Lazy has to come first
```
Comment (dirkgr) [inline tango/integrations/torch/train.py:77]: Preferring `Lazy` solved the problems we were talking about.

**Comment by dirkgr** [comment]
@epwalsh, I think this is ready for review now.

File: CHANGELOG.md [hunk 2]
```diff
@@ -27,6 +27,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fixed a bug in `WandbWorkspace.step_finished()` which led to a `RuntimeError` sometimes while
   caching the result of a step.
 
+### Added
+
+- Added the ability to train straight `Model` instead of just `Lazy[Model]`
```
Comment (epwalsh) [inline CHANGELOG.md:32]: Needs to be moved up to unreleased section.

File: tango/format.py [hunk 3]
```diff
@@ -43,7 +43,7 @@ class Format(Registrable, Generic[T]):
     the result of a :class:`~tango.step.Step`.
     """
 
-    VERSION: int = NotImplemented
+    VERSION: str = NotImplemented
```
Comment (epwalsh) [inline tango/format.py:46]: TIL

File: tango/integrations/torch/train_config.py [hunk 4]
```diff
@@ -74,17 +76,17 @@ class TrainConfig:
 
     log_every: int = 10
     """
-    Controls the frequency of log updates.
+    Controls the frequency of log updates, in number of optimizer steps
```
Comment (epwalsh) [inline tango/integrations/torch/train_config.py:79]: 😉
```suggestion
    Controls the frequency of log updates, in number of optimizer steps.
```

File: tango/integrations/torch/train_config.py [hunk 5]
```diff
@@ -74,17 +76,17 @@ class TrainConfig:
...
     """
 
     checkpoint_every: int = 100
     """
-    Controls the frequency of checkpoints.
+    Controls the frequency of checkpoints, in number of optimizer steps
```
Comment (epwalsh) [inline tango/integrations/torch/train_config.py:84]:
```suggestion
    Controls the frequency of checkpoints, in number of optimizer steps.
```

File: tango/integrations/torch/train_config.py [hunk 6]
```diff
@@ -74,17 +76,17 @@ class TrainConfig:
...
     """
 
     validate_every: Optional[int] = None
     """
-    Controls the frequency of the validation loop.
+    Controls the frequency of the validation loop, in number of optimizer steps
```
Comment (epwalsh) [inline tango/integrations/torch/train_config.py:89]:
```suggestion
    Controls the frequency of the validation loop, in number of optimizer steps.
```

**Review by epwalsh — CHANGES_REQUESTED** [review]
LGTM! Just minor suggestions, and the failing test in CI looks like a quick fix.

File: tango/integrations/torch/train_config.py (hunk 4 above)
Comment (dirkgr) [reply tango/integrations/torch/train_config.py:79]: It's not a complete sentence. No object, no period :-)

Comment (epwalsh) [reply tango/integrations/torch/train_config.py:79]: As an American, I know nothing about grammar.

File: CHANGELOG.md (hunk 2 above)
Comment (dirkgr) [reply CHANGELOG.md]: done

**Review by epwalsh — APPROVED** [review]

**Pull request merged**